src/librustc/ty/instance.rs

```diff
@@ -291,7 +300,16 @@ fn resolve_associated_item<'a, 'tcx>(
             if let Some(_) = tcx.lang_items().clone_trait() {
                 let name = tcx.type_of(def_id);
                 let def = if name == "clone" {
```

Comment:
Wait, pre-existing but... how .. does this work? `name = tcx.type_of(def_id); if name == "clone"`?? Isn't `name` a `Ty<'tcx>`?

Reply:
I guess that the travis failure [answers that question](https://travis-ci.org/rust-lang/rust/builds/338774554#L1884) (i.e., it doesn't work).

Reply:
lol I broke this in a rebase by accident. It used to be `item_name`